cli/compose/convert/compose.go

```diff
@@ -81,3 +84,27 @@ func Networks(namespace Namespace, networks networkMap, servicesNetworks map[str
 
 	return result, externalNetworks
 }
+
+// Secrets converts secrets from the Compose type to the engine API type
+func Secrets(namespace Namespace, secrets map[string]composetypes.SecretConfig) ([]swarm.SecretSpec, error) {
+	result := []swarm.SecretSpec{}
+	for name, secret := range secrets {
+		if secret.External.External {
+			continue
+		}
+
+		data, err := ioutil.ReadFile(secret.File)
```

Comment:
I guess it doesn't have to be set right at the daemon max in case this is ever increased, but I'd prefer to not read in huge files just to crash the client (or potentially cause other issues on the client machine)... it'd be different if we were streaming the file to the daemon.